symfit/core/fit.py

```diff
@@ -1448,10 +1509,11 @@ def r_squared(model, fit_result, data):
     :param data: data with which the fit was performed.
     """
     # First filter out the dependent vars
-    y_is = [data[var] for var in model if var in data]
-    x_is = [value for var, value in data.items() if var.name in model.__signature__.parameters]
+    y_is = [data[var] for var in model.dependent_vars if var in data]
+    x_is = [data[var] for var in model.independent_vars if var in data]
     y_bars = [np.mean(y_i) if y_i is not None else None for y_i in y_is]
-    f_is = model(*x_is, **fit_result.params)
+    f_is = model(*x_is, **fit_result.params)._asdict()
+    f_is = [f_is[var] for var in model.dependent_vars]
```

Comment:
I have the idea this line is a no-op

Reply:
Not if the model has `interdependent_vars`. Since ans really evaluates all components of the model, it will also contain the ones corresponding to the interdependent variables. API wise I think that is fine and consistent with the model basicly being its model_dict, but it means we should ignore them here.

Reply:
aaah.... I missed that edge case indeed :)
Maybe: `f_is = [f_is[var] for var in f_is if var not in model.interdependent_vars]`?

Reply:
I agree that it would be more readable, but I do feel like since there are more steps in between, I can more easily see the ordering go wrong with future bugs by the time we get to the `Ans` object. But I'll add a comment, that should also do the trick.